core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -418,7 +418,14 @@ class KafkaServer(
         kafkaController.startup()
 
         if (config.migrationEnabled) {
-          logger.info("Starting up additional components for ZooKeeper migration")
+          if (config.interBrokerProtocolVersion.isMetadataTransactionSupported) {
+            logger.info("Starting up additional components for ZooKeeper migration")
+          } else {
+            logger.error("Caution! Starting up additional components for ZooKeeper migration without metadata transactions. " +
```

Comment:
Small nit. Can we do the preceding log, but have an additional log at ERROR level saying that we don't support metadata transactions?

Reply:
Yea, fair enough. I wondered the same. Updated